### DISCUSSION TITLE
Optionally implement terraform-provider-value as provider-defined function(s) for Terraform 1.8+

### DISCUSSION THREAD

**Discussion by crw** [pseudo-dynamic/terraform-provider-value#2] [Ideas]
Thank you for supporting the Terraform community with this data provider! If you were using the data provider construct to work around the lack of support in Terraform for external functions, please consider the following new feature of Terraform. 

[Terraform version 1.8](https://github.com/hashicorp/terraform/blob/v1.8.0-beta1/CHANGELOG.md) launches with support of provider-defined functions. It is now possible to implement this functionality as a function, rather than wrap the functionality in a data provider. Provider-based functions have the advantage of a simpler in-line syntax that does not require the data block (or any other resource blocks) to use.

The caveat is that, as provider-defined functions are new to Terraform 1.8, you would need to continue to maintain the current data provider functionality if you wished to continue to support earlier versions of Terraform. 

Please see the [provider-defined functions documentation](https://developer.hashicorp.com/terraform/plugin/framework/functions) to learn how to implement functions in your providers. If you have any questions, please visit the [Terraform topic in our official forum](https://discuss.hashicorp.com/c/terraform-core/27). 

We greatly appreciate your support of the Terraform community with your contributions. **If you do not wish to implement this functionality as a function, please feel free to close this issue**. This issue is simply to raise awareness of the new capability. Thanks again!

**Comment by teneko** [comment]
First of all this is not an issue but a discussion you want to engage and secondly, before you randomly drop "advertisment" it would be nice if you at least spend a few minutes to know whether this feature actually solves the limitation this repo is trying to overcome or not.

**Reply by teneko** [reply]
And since the first words of this feature start with
> Terraform expects that provider-defined functions are implemented without side-effects and as pure functions where given the same input data that they always return the same output.

the feature is nice, but not helpful in this particular case.

**Reply by teneko** [reply]
Please see this [StackOverflow topic](https://stackoverflow.com/questions/73368439/determining-whether-a-value-is-known-at-plan-time) for context.

**Reply by crw** [reply]
Thanks, feel free to close out this topic (if that is a thing, I have not used GitHub discussions much).
